lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -321,14 +315,27 @@ def set_aspect(self, aspect, adjustable=None, anchor=None, share=False):
         --------
         mpl_toolkits.mplot3d.axes3d.Axes3D.set_box_aspect
         """
-        if aspect != 'auto':
+        if aspect not in ['auto', 'equal']:
             raise NotImplementedError(
                 "Axes3D currently only supports the aspect argument "
-                f"'auto'. You passed in {aspect!r}."
+                f"'auto' or 'equal'. You passed in {aspect!r}."
             )
         super().set_aspect(
             aspect, adjustable=adjustable, anchor=anchor, share=share)
 
+        if aspect == 'equal':
+            self.set_box_aspect((1., 1., 1.))
+            v_intervals = np.vstack((self.xaxis.get_view_interval(),
+                                     self.yaxis.get_view_interval(),
+                                     self.zaxis.get_view_interval())).T
+            mean = np.mean(v_intervals, axis=0)
+            delta = np.max(np.max(v_intervals, axis=0) -
+                           np.min(v_intervals, axis=0))
```

Comment:
Can we achieve the same effect using `np.ptp`?

Also, using `axis=0` on the transpose should be the same as using `axis=1` on the original?

Reply:
Yeah, those do clean things up